design/one-pager-stack-cli.md

```diff
@@ -0,0 +1,79 @@
+# Crossplane Stacks CLI
+* Owner: Daniel Suskin (@suskin)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Introduction
+*Why does this document exist?*
+
+We're working on the first version of a user experience for working with crossplane stacks. This document answers the questions of:
+* What are our goals for this version?
+* How do we measure success?
+* What do we think the experience will look like?
+
+## OKRs
+*What are our goals for this version? How do we measure success?*
+
+Objectives:
+* Provide a complete and simple experience for creating stacks from scratch (for a single pattern of creating stacks)
+* The stacks experience can be updated independently of the Crossplane releases
+
+Key Results:
+* Creating a Hello World stack from scratch takes 5 minutes, using the recommended approach
+* Creating and publishing a Hello World stack from scratch takes 10 minutes, using the recommended approach
+* All documentation linking from Crossplane 0.3 to the stacks experience will not need to be changed if we update the stacks experience
+
+## Experience
+*What do we think the experience will look like?*
+
+To create a Hello World stack from scratch:
+* Have access to a Crossplane
+* Install kubebuilder v2 and its prerequisites
+  - go
+  - docker
+  - kustomize (optional)
+* Create a kubebuilder v2 project
+* Add a "Hello World" log line to the kubebuilder controller
+* Install kubectl plugins
+  - `curl -s -o /usr/local/bin/kubectl-crossplane-stack-init /url/directly/to/the/script/somewhere >/dev/null`
+  - More curls until all the scripts are downloaded
+  - `chmod +x /usr/local/bin/kubectl-crossplane-*`
+* Navigate to the project directory
+  - `cd myproject`
+* Run init
+  - `kubectl crossplane stack init mystackgroup/mystackname`
+* Run build
+  - `kubectl crossplane stack build`
+* Run publish
+  - (docker registry credentials should be set up before publishing)
+  - `kubectl crossplane stack publish`
+* Install
+  - `kubectl apply -f config/stack/install.stack.yaml`
+* Validate
+  - `kubectl apply -f config/samples/*instance.yaml`
+  - `kubectl logs mystackname-pod-name`
+
+## FAQ
+### How does configuration of the stack tool work?
+The tool will make assumptions based on a single supported pattern for now, but that will be configurable in the future. When it is configurable, it will also be adjustable at init time.
```

Comment:
do we want to mention what the single supported pattern is right now? we've already introduced the concept of kubebuilder controllers elsewhere in the doc, so this could be as simple as a parenthesis reminder that the pattern is kubebuilder, e.g.
> ... based on a single supported pattern (kubebuilder v3 controllers) for now, ...

Reply:
Yeah, I guess it would make sense to mention that. I was originally thinking that the intent is to support a single pattern, and we just happened to be supporting kubebuilder as the first pattern. I'll play around with some wordings of this.

Reply:
This should be addressed now